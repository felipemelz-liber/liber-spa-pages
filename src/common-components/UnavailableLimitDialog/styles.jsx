import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
