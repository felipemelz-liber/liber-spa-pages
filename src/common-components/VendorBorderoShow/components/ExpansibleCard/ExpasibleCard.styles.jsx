import styled, { css } from 'styled-components';
import { Card as RawCard, Button } from 'liber-components';

export const Card = styled(RawCard)`
  width: 100%;
  min-height: 100px;
  padding: 32px;
  border: solid 1px #cdd7df;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  transition: 300ms linear all;

  ${({ height }) =>
    height &&
    css`
      height: ${height}px;
    `}
`;

export const Container = styled.div`
  width: 100%;
  height: 100%;
  overflow-y: hidden;
`;

export const ExpanseButton = styled(Button)`
  height: 48px;
  border-radius: 24px;
  color: #6287a7;
  background-color: #fff;
  border: solid 1px #cdd7df;
  width: fit-content;
  margin-top: -40px;
  z-index: 5;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;

  :hover {
    background-color: #f2f2f2;
  }

  & > svg {
    width: 20px;
    height: 20px;
    fill: #6287a7;
    margin-bottom: 2px;
    margin-right: 14px;
    transition: 300ms linear all;

    ${({ open }) =>
      open &&
      css`
        transform: rotate(180deg);
      `}
  }
`;

export const ButtonContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;
