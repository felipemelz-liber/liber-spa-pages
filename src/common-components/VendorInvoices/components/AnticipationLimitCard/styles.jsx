import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
