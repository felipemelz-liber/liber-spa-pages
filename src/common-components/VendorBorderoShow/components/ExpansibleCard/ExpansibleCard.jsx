import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import Icon from '@mdi/react';
import { mdiArrowDown } from '@mdi/js';
import { Card, Container, ExpanseButton, ButtonContainer } from './ExpasibleCard.styles';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

const ExpansibleCard = ({ children, buttonText, collapsedHeight, showButton, isAdmin }) => {
  const [open, setOpen] = React.useState(true);
  const [height, setHeight] = useState(undefined);

  const ref = useRef(null);

  useEffect(() => {
    setHeight(ref.current.clientHeight);
    setOpen(false);
  }, []);

  const toggleOpen = () => {
    dispatchDataLayerEvent('showBorderoMostrarConfirmacaoPagamento', isAdmin, {
      mostrarConfirmacaoPagamento: !open,
    });

    setOpen(!open);
  };

  const { open: openText, closed: closedText } = buttonText;

  return (
    <>
      <Card height={open ? height : collapsedHeight} ref={ref}>
        <Container>{children}</Container>
      </Card>
      {showButton && (
        <ButtonContainer>
          <ExpanseButton open={open} onClick={toggleOpen}>
            <Icon path={mdiArrowDown} />
            {open ? openText : closedText}
          </ExpanseButton>
        </ButtonContainer>
      )}
    </>
  );
};

ExpansibleCard.propTypes = {
  children: PropTypes.node,
  buttonText: PropTypes.shape({ closed: PropTypes.string, open: PropTypes.string }),
  collapsedHeight: PropTypes.number,
  showButton: PropTypes.bool,
  isAdmin: PropTypes.bool,
};

ExpansibleCard.defaultProps = {
  children: <div />,
  buttonText: { closed: 'Expandir', open: 'Ocultar' },
  collapsedHeight: 100,
  showButton: true,
  isAdmin: false,
};

export default ExpansibleCard;
