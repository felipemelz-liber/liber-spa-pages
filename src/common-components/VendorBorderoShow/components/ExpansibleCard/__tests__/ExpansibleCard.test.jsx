import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ExpansibleCard from '../ExpansibleCard';
import { ExpanseButton } from '../ExpasibleCard.styles';

describe('ExpansibleCard component test', () => {
  configure({ adapter: new Adapter() });

  const props = {
    children: <div>Mock content</div>,
    buttonText: { closed: 'Click to expand', open: 'Click to collapse' },
    collapsedHeight: 300,
  };

  it('should render correctly without props', () => {
    const wrapper = shallow(<ExpansibleCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<ExpansibleCard {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call setOpen with the oposite of open when clicking on expanse button', () => {
    const mockOpen = false;
    const mockSetOpen = sinon.spy();
    sinon.stub(React, 'useState').callsFake(() => [mockOpen, mockSetOpen]);

    const wrapper = shallow(<ExpansibleCard {...props} />);
    const expanseButton = wrapper.find(ExpanseButton);
    expanseButton.simulate('click');

    expect(mockSetOpen.calledWith(true)).toBeTruthy();
  });
});
