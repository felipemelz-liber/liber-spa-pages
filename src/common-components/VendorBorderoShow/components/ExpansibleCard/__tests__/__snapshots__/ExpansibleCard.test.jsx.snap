// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansibleCard component test should render correctly with props 1`] = `
<Fragment>
  <Styled(Cards__Card)>
    <styled.div>
      <div>
        Mock content
      </div>
    </styled.div>
  </Styled(Cards__Card)>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      open={true}
    >
      <Icon
        color="currentColor"
        horizontal={false}
        path="M11,4H13V16L18.5,10.5L19.92,11.92L12,19.84L4.08,11.92L5.5,10.5L11,16V4Z"
        rotate={0}
        size={null}
        spin={false}
        vertical={false}
      />
      Click to collapse
    </Styled(Buttons__Button)>
  </styled.div>
</Fragment>
`;

exports[`ExpansibleCard component test should render correctly without props 1`] = `
<Fragment>
  <Styled(Cards__Card)>
    <styled.div>
      <div />
    </styled.div>
  </Styled(Cards__Card)>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      open={true}
    >
      <Icon
        color="currentColor"
        horizontal={false}
        path="M11,4H13V16L18.5,10.5L19.92,11.92L12,19.84L4.08,11.92L5.5,10.5L11,16V4Z"
        rotate={0}
        size={null}
        spin={false}
        vertical={false}
      />
      Ocultar
    </Styled(Buttons__Button)>
  </styled.div>
</Fragment>
`;
